.github/workflows/ci.yml

```diff
@@ -23,14 +23,16 @@ jobs:
           key: yarn-${{ hashFiles('yarn.lock') }}
           restore-keys: |
             yarn-
-      - name: Update Yarn cache
+      - name: Check Yarn dedupe
         if: steps.yarn-cache.outputs.cache-hit != 'true'
+        run: |
+          yarn dedupe --check
+      - name: Check or update Yarn cache
         env:
           YARN_ENABLE_SCRIPTS: false # disable post-install scripts
           YARN_NODE_LINKER: pnp # use pnp linker for better performance: it's meant to update yarn cache only
         run: |
-          yarn dedupe --check
-          yarn install --immutable
+          yarn install --immutable --skip-builds
```

Comment:
Note that since we have run `yarn install --immutable --skip-builds` unconditionally, GitHub CI now update the cache even if the `yarn.lock` is not changed and cache was hit. It adds 5 extra seconds to current CI: https://github.com/babel/babel/pull/12508/checks?check_run_id=1557856427#step:13:1
 
https://github.com/actions/cache/pull/489 will eventually solve this issue.